algebra/tests/quat/various.c

```diff
@@ -1280,3 +1312,144 @@ TEST_GROUP_RUNNER(group_quat_uvec2uvec)
 	RUN_TEST_CASE(group_quat_uvec2uvec, quat_uvec2uvec_nearlyAntiparallel);
 	RUN_TEST_CASE(group_quat_uvec2uvec, quat_uvec2uvec_nearlyParallel);
 }
+
+
+/* ##############################################################################
+ * ---------------------        quat_frameRot tests       -----------------------
+ * ############################################################################## */
+
+
+TEST_GROUP(group_quat_frameRot);
+
+
+TEST_SETUP(group_quat_frameRot)
+{
+}
+
+
+TEST_TEAR_DOWN(group_quat_frameRot)
+{
+}
+
+
+TEST(group_quat_frameRot, quat_frameRot_baseRot)
+{
+	const vec_t versX = { .x = 1.0f, .y = 0.0f, .z = 0.0f };
+	const vec_t versY = { .x = 0.0f, .y = 1.0f, .z = 0.0f };
+	const vec_t versZ = { .x = 0.0f, .y = 0.0f, .z = 1.0f };
+	const vec_t negVersY = { .x = 0.0f, .y = -1.0f, .z = 0.0f };
+	quat_t q, expected, helpQ;
+
+	/* Rotating frame of reference 90 degrees along x-axis */
+	quat_rotQuat(&versX, M_PI / 2, &expected);
+	quat_rotQuat(&versX, M_PI / 4, &helpQ);
+
+	quat_frameRot(&versX, &versY, &versX, &versZ, &q, &helpQ);
+
+	TEST_ASSERT_EQUAL_QUAT(expected, q);
+
+	/* Rotating frame of reference 90 degrees along y-axis */
+	quat_rotQuat(&versY, M_PI / 2, &expected);
+	quat_rotQuat(&versY, M_PI / 4, &helpQ);
```

Comment:
Lets use precalculated values :)
```suggestion
	quat_rotQuat(&versY, M_PI_2, &expected);
	quat_rotQuat(&versY, M_PI_4, &helpQ);
```